modules/contact/src/actions/ContactAction.C

```diff
@@ -411,8 +450,26 @@ ContactAction::act()
   if (_current_task == "add_user_object")
   {
     auto var_params = _factory.getValidParams("NodalArea");
-    var_params.set<std::vector<BoundaryName>>("boundary") =
-        getParam<std::vector<BoundaryName>>("secondary");
+
+    // Get secondary_boundary_vector from possibly updated set from the
+    // ContactAction constructor cleanup
+    const auto actions = _awh.getActions<ContactAction>();
+
+    std::size_t all_action_pairs_size = 0;
+    for (const auto * const action : actions)
+      all_action_pairs_size += action->_boundary_pairs.size();
+
+    std::vector<BoundaryName> secondary_boundary_vector(all_action_pairs_size);
+
+    std::size_t i = 0;
+    for (const auto * const action : actions)
+      for (const auto j : index_range(action->_boundary_pairs))
+      {
+        secondary_boundary_vector[i] = action->_boundary_pairs[j].second;
+        i++;
+      }
```

Comment:
My suggestion here would be

```suggestion
    std::vector<BoundaryName> secondary_boundary_vector();
    for (const auto * const action : actions)
      for (const auto j : index_range(action->_boundary_pairs))
        secondary_boundary_vector.push_back(action->_boundary_pairs[j].second);
```

as this only runs on setup and pre-sizing the vector seems like an unnecessary optimization that triples the code size here.

Reply:
Sure.